gfx.h

```diff
@@ -1402,6 +1402,24 @@ class GfxInternal
         device_->QueryInterface(IID_PPV_ARGS(&mesh_device_));
         SetDebugName(device_, "gfx_Device");
 
+        if((flags & kGfxCreateContextFlag_EnableDebugLayer) != 0)
+        {
+            ID3D12InfoQueue1 *debugCallback;
+            device_->QueryInterface(IID_PPV_ARGS(&debugCallback));
+            DWORD cookie;
+            D3D12MessageFunc callback = [](D3D12_MESSAGE_CATEGORY Category, D3D12_MESSAGE_SEVERITY Severity,
+                                            D3D12_MESSAGE_ID ID, LPCSTR pDescription, void *pContext) {
+                if(Severity <= D3D12_MESSAGE_SEVERITY_ERROR)
+                {
+                    GFX_PRINTLN("D3D12 Error: %s", pDescription);
+                    GFX_ASSERT(false);
+                } else if(Severity == D3D12_MESSAGE_SEVERITY_WARNING) {
+                    GFX_PRINTLN("D3D12 Warning: %s", pDescription);
+                }
+            };
+            debugCallback->RegisterMessageCallback(callback, D3D12_MESSAGE_CALLBACK_FLAG_NONE, this, &cookie);
```

Comment:
`debugCallback` can be null here (for instance, by disabling Developer Mode in Windows Developer Settings I believe? or if the Graphics Tools optional feature isn't installed) so we should have a pointer check to avoid crashing.

Reply:
fixed